### PR TITLE
Fix Perf Test Regression

### PR DESCRIPTION
**Pull request by cicoyle**
Reverting the changes from this PR updating actor code: [update master w/ release-1.13 branch changes](https://github.com/dapr/dapr/pull/7599) resulted in the perf tests passing again, specifically the `TestActorReminderRegistrationPerformance` test. 
The changes involve reverting the: json serialization and api level.

### CODE REVIEW

**Comment by cicoyle** [comment]
Just for grins I'll re-run the perf tests

File: tests/perf/actor_reminder/actor_reminder_test.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ const (
 	appName         = "perf-actor-reminder-service"
 
 	// Target for the QPS
-	targetQPS = 56
+	targetQPS = 50
```
Comment (cicoyle) [inline tests/perf/actor_reminder/actor_reminder_test.go:42]: Lowering this to acct for json being made the default, which isn't as performant as using protobufs for storage - the diff is quite noticeable with the target app CPU time. Using json requires the app to marshal and unmarshal the data  increasing the target app CPU time vs using protobufs which significantly lowers the target app CPU time.

This target app CPU time diff btw protobuf & json is: 
JSON: 7m CPU time 
Protobuf: 1m CPU time

**Comment by cicoyle** [comment]
Looks like `TestBulkPubsubPublishGrpcPerformance` might be flaky? I will re-run the perf test since actor reminder registration one is passing now

**Comment by cicoyle** [comment]
Maybe that unit test for placement membership is flaky?

**Comment by ItalyPaleAle** [comment]
@cicoyle you can use `/ok-to-perf actor_reminder` to run just the test you're debugging, avoiding running for 3 hours which burns credit and puts out CO2 for no reason :)

**Comment by cicoyle** [comment]
Looks like there is a panic with `controller-gen` that was recently introduced. This PR confirms a perf regression of ~5 for actor reminder registration - need to look into this still

**Review by mikeee — APPROVED** [review]
Lgtm

**Review by mikeee — APPROVED** [review]
https://github.com/dapr/dapr/blob/81727bab7306e8e0b68393a4d91434b9762b7611/pkg/api/http/responses.go

Nit would be to remove this dead code to make the linter happy

**Comment by cicoyle** [comment]
This was already merged in, so going to close this.

**Pull request closed without merging**